samples/sdl_gamecontroller_event/main.c

```diff
@@ -0,0 +1,56 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  assert(SDL_IsGameController(gameControllerIndex));
+  int gameControllerIndex = 0;
+  assert(SDL_IsGameController(gameControllerIndex));
+  SDL_GameController *gameController = SDL_GameControllerOpen(gameControllerIndex);
+  if (gameController == NULL) {
+    debugPrint("Could not open GameController %i, exiting...", gameControllerIndex);
+    SDL_QuitSubSystem(SDL_INIT_GAMECONTROLLER);
+    XSleep(2000);
+    return -1;
+  }
+
+  debugPrint("GameController name: %s\n", SDL_GameControllerName(gameController));
+  debugPrint("Is GameController  : %s\n", SDL_IsGameController(gameControllerIndex)?"Yes":"No");
+  debugPrint("GameController Map : %s\n",   SDL_GameControllerMapping(gameController));
+  debugPrint("GameControllerEvent: %2i\n", SDL_GameControllerEventState(SDL_QUERY));
+  debugPrint("\n");
+
+  while (true) {
+    SDL_Event event;
+    while (SDL_PollEvent(&event)) {
+      if (event.type == SDL_CONTROLLERBUTTONDOWN) {
+        debugPrint("Button \"%s\" pressed\n",
+                   SDL_GameControllerGetStringForButton((SDL_GameControllerButton)event.cbutton.button));
+      } else if (event.type == SDL_CONTROLLERAXISMOTION) {
+        debugPrint("Axis \"%s\" moved, new value: %i\n",
+                   SDL_GameControllerGetStringForAxis((SDL_GameControllerAxis)event.caxis.axis),
+                   event.caxis.value);
+      }
+    }
+  }
+
+  if (gameController != NULL) {
```

Comment:
This sample can still continue, even if gamecontroller wasn't successfully opened. There just won't be any events.

But it makes the assumption that it will only continue if the gamecontroller was opened successfully. So this check is probably useless, or the exit above is useless.